src/state/layoutSlice.ts

```diff
@@ -16,9 +28,15 @@ const chartLayoutSlice = createSlice({
     setLayout(state, action: PayloadAction<LayoutType>) {
       state.layoutType = action.payload;
     },
+    setContainer(state, action: PayloadAction<HasDomRect | null>) {
+      state.container = castDraft(action.payload);
```

Comment:
You dispatch an HTML element but only type one function?

Reply:
I guess that's to prevent us from doing stupid stuff. But the other properties will still be there

Reply:
The interface is there to make mocks and unit testing easier, and to show which properties of the element is recharts using.